Framework/Algorithms/test/CrossCorrelateTest.h

```diff
@@ -0,0 +1,179 @@
+#ifndef CROSSCORRELATETEST_H_
+#define CROSSCORRELATETEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidHistogramData/LinearGenerator.h"
+#include "MantidAlgorithms/CrossCorrelate.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+using namespace Mantid;
+using namespace Mantid::Algorithms;
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using Mantid::HistogramData::BinEdges;
+using Mantid::HistogramData::CountStandardDeviations;
+
+class CrossCorrelateTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being
+  // created statically This means the constructor isn't called when
+  // running other tests
+  static CrossCorrelateTest *createSuite() { return new CrossCorrelateTest(); }
+  static void destroySuite(CrossCorrelateTest *suite) { delete suite; }
+
+  void testValidInput() {
+    // setup and run the algorithm (includes basic checks)
+    CrossCorrelate alg;
+    const MatrixWorkspace_const_sptr inWS = setupAlgorithm(alg, 2.0, 4.0);
+    const MatrixWorkspace_const_sptr outWS = runAlgorithm(alg, inWS);
+
+    // specific checks
+    const Mantid::MantidVec outX = outWS->readX(0);
+    TS_ASSERT_EQUALS(outX.size(), 3);
+    TS_ASSERT_DELTA(outX[0], -1.0, 1e-6);
+    TS_ASSERT_DELTA(outX[1], 0.0, 1e-6);
+    TS_ASSERT_DELTA(outX[2], 1.0, 1e-6);
+
+    const Mantid::MantidVec outY0 = outWS->readY(0);
+    TS_ASSERT_EQUALS(outY0.size(), 3);
+    TS_ASSERT_DELTA(outY0[0], -0.018902, 1e-6);
+    TS_ASSERT_DELTA(outY0[1], 1.0, 1e-6);
+    TS_ASSERT_DELTA(outY0[2], -0.018902, 1e-6);
+
+    const Mantid::MantidVec outY1 = outWS->readY(1);
+    TS_ASSERT_EQUALS(outY1.size(), 3);
+    TS_ASSERT_DELTA(outY1[0], -0.681363, 1e-6);
+    TS_ASSERT_DELTA(outY1[1], 0.168384, 1e-6);
+    TS_ASSERT_DELTA(outY1[2], 0.456851, 1e-6);
+  }
+
+  // This tests an input X length of 3, which is the minimum the algorithm can
+  // handle
+  void testMinimumInputXLength() {
+    // setup and run the algorithm (includes basic checks)
+    CrossCorrelate alg;
+    const MatrixWorkspace_const_sptr inWS = setupAlgorithm(alg, 2.0, 3.5);
+    const MatrixWorkspace_const_sptr outWS = runAlgorithm(alg, inWS);
+
+    // specific checks
+    const Mantid::MantidVec outX = outWS->readX(0);
```

Comment:
Ah, I'm `using namespace Mantid` anyway, so I can just use `const MantidVec`. I would prefer that, unless there is a convention to use `auto`?